subsys/bluetooth/controller/ll_sw/ull_llcp_chmu.c

```diff
@@ -0,0 +1,322 @@
+		ctx->state = LP_CHMU_STATE_WAIT_TX_CHAN_MAP_IND;
+	} else {
+		llcp_rr_set_incompat(conn, INCOMPAT_RESOLVABLE);
+
+		/* TODO Hardcoded instant delta */
+		ctx->data.chmu.instant = lp_event_counter(conn) + CHMU_INSTANT_DELTA;
+
+		lp_chmu_tx(conn, ctx);
+
+		ctx->state = LP_CHMU_STATE_WAIT_INSTANT;
+	}
+}
+
+static void lp_chmu_st_idle(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	switch (evt) {
+	case LP_CHMU_EVT_RUN:
+		lp_chmu_send_channel_map_update_ind(conn, ctx, evt, param);
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+static void lp_chmu_st_wait_tx_chan_map_ind(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+					    void *param)
+{
+	switch (evt) {
+	case LP_CHMU_EVT_RUN:
+		lp_chmu_send_channel_map_update_ind(conn, ctx, evt, param);
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+static void lp_chmu_check_instant(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				  void *param)
+{
+	uint16_t event_counter = lp_event_counter(conn);
+
+	if (is_instant_reached_or_passed(ctx->data.chmu.instant, event_counter)) {
+		llcp_rr_set_incompat(conn, INCOMPAT_NO_COLLISION);
+		lp_chmu_complete(conn, ctx, evt, param);
+	}
+}
+
+static void lp_chmu_st_wait_instant(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				    void *param)
+{
+	switch (evt) {
+	case LP_CHMU_EVT_RUN:
+		lp_chmu_check_instant(conn, ctx, evt, param);
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+static void lp_chmu_execute_fsm(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				void *param)
+{
+	switch (ctx->state) {
+	case LP_CHMU_STATE_IDLE:
+		lp_chmu_st_idle(conn, ctx, evt, param);
+		break;
+	case LP_CHMU_STATE_WAIT_TX_CHAN_MAP_IND:
+		lp_chmu_st_wait_tx_chan_map_ind(conn, ctx, evt, param);
+		break;
+	case LP_CHMU_STATE_WAIT_INSTANT:
+		lp_chmu_st_wait_instant(conn, ctx, evt, param);
+		break;
+	default:
+		/* Unknown state */
+		LL_ASSERT(0);
+	}
+}
+
+void llcp_lp_chmu_init_proc(struct proc_ctx *ctx)
+{
+	ctx->state = LP_CHMU_STATE_IDLE;
+}
+
+void llcp_lp_chmu_run(struct ll_conn *conn, struct proc_ctx *ctx, void *param)
+{
+	lp_chmu_execute_fsm(conn, ctx, LP_CHMU_EVT_RUN, param);
+}
+#endif /* CONFIG_BT_CENTRAL */
+
+#if defined(CONFIG_BT_PERIPHERAL)
+/*
+ * LLCP Remote Procedure Channel Map Update FSM
+ */
+
+/* TODO should go into some utils file */
+static uint16_t rp_event_counter(struct ll_conn *conn)
+{
+	struct lll_conn *lll = &conn->lll;
+
+	/* Calculate current event counter */
+	return lll->event_counter + lll->latency_prepare;
+}
+
+static void rp_chmu_complete(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	ull_conn_chan_map_set(conn, ctx->data.chmu.chm);
+	llcp_rr_complete(conn);
+	ctx->state = RP_CHMU_STATE_IDLE;
+}
+
+static void rp_chmu_st_idle(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	/* TODO */
```

Comment:
Still valid TODO? 

Empty TODOs usually aren't very hopeful.

Reply:
We have a See issue #15256
One of the remaining action items is to go through all TODOs and evaluate what is to be done with these